src/app/create-report/create-report.component.ts

```diff
@@ -504,6 +523,21 @@ export class CreateReportComponent implements OnInit, AfterViewInit {
       );
   }
 
+  private getTwitterApiVersion() {
+    this.socialMediaItemsService.getTwitterApiVersion().subscribe((version:TwitterApiVersion) => {
+      if(version === TwitterApiVersion.ESSENTIAL_OR_ELEVATED_V2) {
+        this.useEssentialOrElevatedV2 = true;
+      } else {
+        this.useEssentialOrElevatedV2 = true;
+      }
+    },
+      _error => {
+        this.useEssentialOrElevatedV2 = false;
+      }
+    )
+    this.dateDropdownOptions = this.useEssentialOrElevatedV2 ?  this.essentialDateDropdownOptions : this.fullDateDropdownOptions
```

Comment:
I take it this is like `finally` block i.e. it will execute whether the call above was a success or error.

Reply:
No. It will only execute on error. Alternatively, I can use a `try-catch` to handle the same.

Reply:
What I mean is: fetch version; if successful, set to fetched version. on error, set to default version (I'm assuming enterprise i.e. `useEssentialOrElevatedV2 = false`)

Here is the crucial part, whether the API call was successful or not, at the end of this call, we **must** update `dateDropdownOptions` to reflect the new value of `useEssentialOrElevatedV2`.

Reply:
@kilemensi The current implementation handles this. By default `useEssentialorElavatedV2` is false by default, and will only update `dateDropdownOptions` if the API Call was successful

Reply:
I don't know how `subscribe` works @kelvinkipruto but if you're sure, cool. My concern was looking at it, https://github.com/CodeForAfrica/harassment-manager/blob/14270012b759f83c57aa9fb56108e07a0ea279a3/src/app/create-report/create-report.component.ts#L527-L529 appears to be what we do when the call was successful. I don't see us updating `dateDropdownOptions` there.

Reply:
@kilemensi Thanks for this. I had missed out. Thinking further about it, we do not need the error handling as we'll only need to update `dateDropdownOptions` if api version request was successful.